python/dgl/graphbolt/copy_to.py

```diff
@@ -0,0 +1,42 @@
+"""Graph Bolt CUDA-related Data Pipelines"""
+
+import torch
+from torchdata.datapipes.iter import IterDataPipe
+
+from ..utils import recursive_apply
+
+
+def _to(x, device):
+    return x.to(device) if hasattr(x, "to") else x
+
+
+class CopyTo(IterDataPipe):
```

Comment:
Why not also use Mapper?

Reply:
I think the current implementation is more straightforward.  If I use a `Mapper` then I'll have to write it like:
```python
def __init__(self, dp, device):
    self.device = device
    super().__init__(dp, self.map_func)
def map_func(self, data):
    return recursive_apply(data, _to, self.device)
```

Reply:
This is 5 lines compare to existing 8 lines of code, it is not bad.

Reply:
But I don't think it's as readable as 8 lines of code.  A for loop is cleaner than passing in a function IMO.